test/ValidatorSet/ValidatorSet.test.ts

```diff
@@ -356,18 +354,110 @@ describe("ValidatorSet", function () {
+          VALIDATOR_STATUS.Whitelisted
+        );
+      });
+
+      it("owner should not whitelist user that is already whitelisted", async function () {
+        const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.be.revertedWith("Previously whitelisted.");
+
+        expect((await validatorSet.validators(this.signers.validators[0].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
       });
 
       it("should be able to remove from whitelist", async function () {
         const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
 
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[1].address, this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
+        expect((await validatorSet.validators(this.signers.validators[3].address)).status).to.be.equal(
+          VALIDATOR_STATUS.None
+        );
+
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+      });
+
+      it("should revert if we remove from whitelist twice", async function () {
+        const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[1].address, this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
         await expect(
           validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
         ).to.not.be.reverted;
 
-        expect((await validatorSet.validators(this.signers.validators[1].address)).whitelisted).to.be.false;
+        await expect(
+          validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
+        ).to.be.revertedWith("Must be whitelisted.");
+
+        expect((await validatorSet.validators(this.signers.validators[3].address)).status).to.be.equal(
```

Comment:
Same as the previous comments - we don't have to verify something that is already verified to work.